flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobClient.java

```diff
@@ -593,9 +594,7 @@ private void sendPutHeader(OutputStream outputStream, JobID jobID, String key) t
 	 *         the BLOB server or if the BLOB server cannot delete the file
 	 */
 	public void delete(BlobKey key) throws IOException {
-		if (key == null) {
-			throw new IllegalArgumentException("BLOB key must not be null");
-		}
+		checkArgument(key != null, "BLOB key must not be null.");
```

Comment:
Yes, but it would be cleaner and less surprising for future users. However I have no strong feelings about it.